language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java

```diff
@@ -117,6 +118,18 @@ public static Hover getHover(HoverContext context) {
         return hoverObj;
     }
 
+    private static Optional<Symbol> getSymbolAtCursor(HoverContext context, SemanticModel semanticModel,
+                                                      Document srcFile, LinePosition linePosition) {
+        NonTerminalNode cursor = context.getNodeAtCursor();
+        if (cursor.kind() == SyntaxKind.LIST || cursor.kind() == SyntaxKind.PARENTHESIZED_ARG_LIST
+                || cursor.kind() == SyntaxKind.SIMPLE_NAME_REFERENCE
+                && cursor.parent().kind() == SyntaxKind.CLIENT_RESOURCE_ACCESS_ACTION) {
+            return semanticModel.symbol(cursor.parent());
+        } else {
+            return semanticModel.symbol(srcFile, linePosition);
+        }
```

Comment:
Do we need the `else` statement here?

Reply:
The hover provider earlier used this approach to get the symbol, but in our case we didn't get the correct symbol from this approach hence we had to special case this. If we change the approach to get the symbol using the node it would affect many test cases because the line position approach used by the semantic model and the node at cursor approach use two different methods.